Loader component reports issues when consumer application tries to build

### 💡 Description

The Loader component is reporting multiple issues when it is used in another application even though the wds-react build does not report any build issues.

Here are the error reports currently being generated from our frontend application:

```
src/pages/instruments/detail.tsx:195:12 - error TS2786: 'Loader' cannot be used as a JSX component.
  Its type '(props: LoaderProps) => ReactElement<any, string | JSXElementConstructor<any>>' is not a valid JSX element type.

195           <Loader variant="indeterminate"/>
 
...

src/pages/investigations/detail.tsx:347:12 - error TS2786: 'Loader' cannot be used as a JSX component.
  Its type '(props: LoaderProps) => ReactElement<any, string | JSXElementConstructor<any>>' is not a valid JSX element type.

347           <Loader variant="indeterminate"/>

...

src/pages/investigations/index.tsx:297:14 - error TS2322: Type '{}' is not assignable to type 'IntrinsicAttributes & LoaderProps'.

297             <Loader />
                 ~~~~~~

src/pages/investigations/index.tsx:297:14 - error TS2786: 'Loader' cannot be used as a JSX component.
  Its type '(props: LoaderProps) => ReactElement<any, string | JSXElementConstructor<any>>' is not a valid JSX element type.
    Type '(props: LoaderProps) => ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type '(props: any, deprecatedLegacyContext?: any) => ReactNode'.
      Type 'ReactElement<any, string | JSXElementConstructor<any>>' is not assignable to type 'ReactNode'.
        Property 'children' is missing in type 'ReactElement<any, string | JSXElementConstructor<any>>' but required in type 'ReactPortal'.

297             <Loader />
```

### ⚔️ Parent Epic / Related Tickets

_No response_

--- a/src/components/Loader/Loader.tsx
+++ b/src/components/Loader/Loader.tsx
@@ -1,7 +1,6 @@
 import CircularProgress, { CircularProgressProps } from '@mui/material/CircularProgress';
 import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
-import { ReactElement } from "react";
 
 type DeterminateLoaderProps = {
   variant: 'determinate'
@@ -48,14 +47,10 @@ function CircularProgressWithLabel(
  */
 export const Loader = (props: LoaderProps) => {
 
-  let progressComponent:ReactElement;
   if( props.variant == "determinate") {
-    progressComponent = <CircularProgressWithLabel value={props.value}/>
-  } else {
-    progressComponent = <CircularProgress />
+    return <CircularProgressWithLabel value={props.value}/>
   }
+  
+  return <CircularProgress />
 
-  return (
-    progressComponent
-  );
 };

--- a/src/components/Loader/Loader.tsx
+++ b/src/components/Loader/Loader.tsx
@@ -2,17 +2,9 @@ import CircularProgress, { CircularProgressProps } from '@mui/material/CircularP
 import Typography from '@mui/material/Typography';
 import Box from '@mui/material/Box';
 
-type DeterminateLoaderProps = {
-  variant: 'determinate'
-  /** Progress value to display */
-  value: number;
-};
-
-type IndeterminateLoaderProps = {
-  variant: 'indeterminate'
+type LoaderProps = {
+  value?: number;
 }
-
-type LoaderProps = DeterminateLoaderProps | IndeterminateLoaderProps;
 
 function CircularProgressWithLabel(
   props: CircularProgressProps & { value: number },
@@ -43,12 +35,13 @@ function CircularProgressWithLabel(
 }
 
 /**
- * A basic determinate or indeterminate loader
+ * A basic loader that can display either an indeterminate
+ * or determinate progress component if a value is passed.
  */
-export const Loader = (props: LoaderProps) => {
+export const Loader = ({value}: LoaderProps) => {
 
-  if( props.variant == "determinate") {
-    return <CircularProgressWithLabel value={props.value}/>
+  if( value !== undefined) {
+    return <CircularProgressWithLabel value={value}/>
   }
   
   return <CircularProgress />

